File: frontend/src/types/stock-analysis-overview.ts
export type StockAnalysisOverviewTaskItem = {
  task_id: number;
  thread_id: number;
  thread_title: string;
  title: string;
  summary: string;
  priority: string;
  status: string;
  task_type: string;
  suggested_action?: string | null;
  reason?: string | null;
};

export type StockAnalysisOverviewActionQueueItem = {
  kind: string;
  thread_id: number;
  title: string;
  summary: string;
  priority: string;
  reason: string;
  suggested_action: string;
  target_ref?: string | null;
};

export type StockAnalysisResearchQualitySummary = {
  effective_count: number;
  mixed_count: number;
  under_evidenced_count: number;
  over_researched_count: number;
  confirmed_count: number;
  partially_confirmed_count: number;
  unclear_count: number;
  contradicted_count: number;
};

export type StockAnalysisPlanningProfileSummary = {
  balanced_count: number;
  refresh_first_count: number;
  compare_first_count: number;
  internal_first_count: number;
  external_confirm_first_count: number;
  lightweight_research_count: number;
};

export type StockAnalysisRecentFeedbackSummary = {
  total_feedback_count: number;
  effective_thread_count: number;
  high_conflict_thread_count: number;
  needs_refresh_thread_count: number;
  follow_up_required_thread_count: number;
  latest_alignment_statuses: string[];
};

export type StockAnalysisThreadOverviewItem = {
  thread_id: number;
  title: string;
  focus_type: string;
  updated_at: string;
  context_count: number;
  compare_target_count: number;
  open_task_count: number;
  high_priority_task_count: number;
  active_memory_available: boolean;
  active_compression_available: boolean;
  refresh_recommended_count: number;
  stale_context_count: number;
  latest_planning_profile?: string | null;
  latest_conflict_level?: string | null;
  latest_validation_status?: string | null;
  latest_feedback_alignment_status?: string | null;
  latest_process_quality_status?: string | null;
  thread_health_status: string;
  thread_health_score: number;
  thread_health_reason: string;
  headline_summary: string;
  next_best_action: string;
  next_best_action_reason: string;
};

export type StockAnalysisOverview = {
  generated_at: string;
  available: boolean;
  summary: string;
  thread_overview_items: StockAnalysisThreadOverviewItem[];
  high_priority_tasks: StockAnalysisOverviewTaskItem[];
  high_conflict_threads: StockAnalysisThreadOverviewItem[];
  refresh_needed_threads: StockAnalysisThreadOverviewItem[];
  research_quality_summary: StockAnalysisResearchQualitySummary;
  planning_profile_summary: StockAnalysisPlanningProfileSummary;
  recent_feedback_summary: StockAnalysisRecentFeedbackSummary;
  action_queue: StockAnalysisOverviewActionQueueItem[];
  empty_message?: string | null;
};

File: python/valuecell/server/api/schemas/stock_analysis_overview.py
from __future__ import annotations

from datetime import datetime
from pydantic import BaseModel, Field


class StockAnalysisOverviewTaskItemData(BaseModel):
    task_id: int
    thread_id: int
    thread_title: str
    title: str
    summary: str
    priority: str
    status: str
    task_type: str
    suggested_action: str | None = None
    reason: str | None = None


class StockAnalysisOverviewActionQueueItemData(BaseModel):
    kind: str
    thread_id: int
    title: str
    summary: str
    priority: str
    reason: str
    suggested_action: str
    target_ref: str | None = None


class StockAnalysisResearchQualitySummaryData(BaseModel):
    effective_count: int = 0
    mixed_count: int = 0
    under_evidenced_count: int = 0
    over_researched_count: int = 0
    confirmed_count: int = 0
    partially_confirmed_count: int = 0
    unclear_count: int = 0
    contradicted_count: int = 0


class StockAnalysisPlanningProfileSummaryData(BaseModel):
    balanced_count: int = 0
    refresh_first_count: int = 0
    compare_first_count: int = 0
    internal_first_count: int = 0
    external_confirm_first_count: int = 0
    lightweight_research_count: int = 0


class StockAnalysisRecentFeedbackSummaryData(BaseModel):
    total_feedback_count: int = 0
    effective_thread_count: int = 0
    high_conflict_thread_count: int = 0
    needs_refresh_thread_count: int = 0
    follow_up_required_thread_count: int = 0
    latest_alignment_statuses: list[str] = Field(default_factory=list)


class StockAnalysisThreadOverviewItemData(BaseModel):
    thread_id: int
    title: str
    focus_type: str
    updated_at: datetime
    context_count: int = 0
    compare_target_count: int = 0
    open_task_count: int = 0
    high_priority_task_count: int = 0
    active_memory_available: bool = False
    active_compression_available: bool = False
    refresh_recommended_count: int = 0
    stale_context_count: int = 0
    latest_planning_profile: str | None = None
    latest_conflict_level: str | None = None
    latest_validation_status: str | None = None
    latest_feedback_alignment_status: str | None = None
    latest_process_quality_status: str | None = None
    thread_health_status: str
    thread_health_score: int
    thread_health_reason: str
    headline_summary: str
    next_best_action: str
    next_best_action_reason: str


class StockAnalysisOverviewData(BaseModel):
    generated_at: datetime
    available: bool
    summary: str
    thread_overview_items: list[StockAnalysisThreadOverviewItemData] = Field(
        default_factory=list
    )
    high_priority_tasks: list[StockAnalysisOverviewTaskItemData] = Field(
        default_factory=list
    )
    high_conflict_threads: list[StockAnalysisThreadOverviewItemData] = Field(
        default_factory=list
    )
    refresh_needed_threads: list[StockAnalysisThreadOverviewItemData] = Field(
        default_factory=list
    )
    research_quality_summary: StockAnalysisResearchQualitySummaryData = Field(
        default_factory=StockAnalysisResearchQualitySummaryData
    )
    planning_profile_summary: StockAnalysisPlanningProfileSummaryData = Field(
        default_factory=StockAnalysisPlanningProfileSummaryData
    )
    recent_feedback_summary: StockAnalysisRecentFeedbackSummaryData = Field(
        default_factory=StockAnalysisRecentFeedbackSummaryData
    )
    action_queue: list[StockAnalysisOverviewActionQueueItemData] = Field(
        default_factory=list
    )
    empty_message: str | None = None
